src/App.jsx

```diff
@@ -14,45 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goodsArr, setGoodsArr] = useState(goodsFromServer);
+  const [sortValue, setSortValue] = useState('');
+  const [reverseValue, setReverseValue] = useState(false);
+  const [resetValue, setResetValue] = useState(false);
+
+  const SORT_BY_ALPHABETICALLY = 'alphabetically';
+  const SORT_BY_LENGTH = 'length';
+
+  const sortBy = (value) => {
+    if (value === SORT_BY_ALPHABETICALLY) {
+      if (reverseValue) {
+        setGoodsArr([...goodsArr].sort((a, b) => b.localeCompare(a)));
```

Comment:
you can use previous value of state, and better will be create sort with if else, instead of if/else and two sorts

Reply:
still not fixed

Reply:
<img width="710" alt="image" src="https://github.com/mate-academy/react_list-of-goods-js/assets/70220667/e2f9c08e-6f00-4860-847b-7e1fc5584c39">

Reply:
please use this arctice as example https://legacy.reactjs.org/docs/hooks-reference.html#usestate